Content.Server/Cluwne/CluwneSystem.cs

```diff
@@ -0,0 +1,121 @@
+        SubscribeLocalEvent<CluwneComponent, ComponentStartup>(OnComponentStartup);
+    }
+
+    private void OnMobState(EntityUid uid, CluwneComponent component, MobStateChangedEvent args)
+    {
+        if (args.NewMobState == MobState.Dead)
+        {
+            RemComp<CluwneComponent>(uid);
+            var damageSpec = new DamageSpecifier(_prototypeManager.Index<DamageGroupPrototype>("Genetic"), 300);
+            _damageableSystem.TryChangeDamage(uid, damageSpec);
+        }
+    }
+    /// <summary>
+    /// Gives target cluwne scream, backwards accent, makes clumsy, inserts a popup message and gives cluwne clothing.
+    /// </summary>
+    private void OnComponentStartup(EntityUid uid, CluwneComponent component, ComponentStartup args)
+    {
+        var meta = MetaData(uid);
+        var name = meta.EntityName;
+
+        EnsureComp<BackwardsAccentComponent>(uid);
+
+        var vocal = EnsureComp<VocalComponent>(uid);
+        var scream = new SoundCollectionSpecifier("CluwneScreams");
+        vocal.FemaleScream = scream;
+        vocal.MaleScream = scream;
+
+        EnsureComp<ClumsyComponent>(uid);
+
+        _popupSystem.PopupEntity(Loc.GetString("cluwne-transform", ("target", uid)), uid, PopupType.LargeCaution);
+        _audio.PlayPvs(component.SpawnSound, uid);
+
+        meta.EntityName = Loc.GetString("cluwne-name-prefix", ("target", name));
+
+        SetOutfitCommand.SetOutfit(uid, "CluwneGear", EntityManager);
+    }
+    /// <summary>
+    /// Makes cluwne do a random emote. Falldown and horn, twitch and honk, giggle.
+    /// </summary>
+    private void DoGiggle(EntityUid uid, CluwneComponent component)
+    {
+        if (component.LastGiggleCooldown > 0)
+            return;
+
+            if (_robustRandom.Prob(0.2f))
+            { 
+                _audio.PlayPvs(component.KnockSound, uid);
+                _stunSystem.TryParalyze(uid, TimeSpan.FromSeconds(component.ParalyzeTime), true);
+            }
+
+            if (_robustRandom.Prob(0.3f))
+            {
+                _audio.PlayPvs(component.Giggle, uid);
+            }
+
+            else
+            {
+                _popupSystem.PopupEntity(Loc.GetString("cluwne-twitch", ("target", Identity.Entity(uid, EntityManager))), uid);
+                _audio.PlayPvs(component.SpawnSound, uid);
+            }
+
+        component.LastGiggleCooldown = component.GiggleCooldown;
+         
+    }
+
+    public override void
+         Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var cluwnecomp in EntityQuery<CluwneComponent>())
+        {
+            cluwnecomp.LastGiggleCooldown -= frameTime;
+
+
+            if (_timing.CurTime <= cluwnecomp.GiggleGoChance) 
+                continue;
+
+            cluwnecomp.GiggleGoChance += TimeSpan.FromSeconds(cluwnecomp.RandomGiggleAttempt);
+
+            if (!_robustRandom.Prob(cluwnecomp.GiggleChance))
+                continue;
+
```

Comment:
Accumulators are deprecated but this code should be shared with zombies anyway.

Reply:
I cut the amount of code here needed in half for cluwne system to function. I also started work on a randomemotesystem but im still unsure if it is needed. I thought i would get your opinion on what i have in cluwne system, if it still needs the randomemotesystem, what i have in the randomemotesystem and which bits of code are still accumulators as im still usnure on it.